active_learning/callbacks/confidences.py

```diff
@@ -20,3 +20,5 @@ def on_predict_end(self, *args, **kwargs):
         outputs = nn.Softmax(dim=1)(torch.cat(self._outputs))
 
         self.predict_outputs = outputs.cpu().numpy()
+        # Reset outputs - just one initialization of callback is needed
+        self._outputs = []
```

Comment:
What are consequences of leaving this unchanged?

I am asking since it is code from outside our package so potential marge conflict :stuck_out_tongue:

Reply:
In basic implemetation this callback is called in each iteration of main `while` loop. It makes ` self._outputs = []` and then appends predict outputs to ` self._outputs`. I initialize this callback once - in base class `__init__`. To make sure that `self._outputs` doesn't contain previous data, I need to clean it at the end of prediction. Why? Because of  
```python
 outputs = nn.Softmax(dim=1)(torch.cat(self._outputs))
 self.predict_outputs = outputs.cpu().numpy()
```
`self.predict_outputs ` is used in `active_learning_flows` to get `self.confidences` values. 
IMO it isn't as invasive as making return statement.